arangod/IResearch/IResearchCommon.cpp

```diff
@@ -27,16 +27,7 @@
 namespace arangodb {
 namespace iresearch {
 
-LogTopic TOPIC(std::string{StaticStrings::DataSourceType}, LogLevel::INFO);
-
-arangodb::LogicalDataSource::Type const& dataSourceType() {
-  static auto& type =
-      arangodb::LogicalDataSource::Type::emplace(StaticStrings::DataSourceType);
-
-  return type;
-}
-
-arangodb::LogTopic& logTopic() { return TOPIC; }
+LogTopic TOPIC{std::string{StaticStrings::DataSourceType}, LogLevel::INFO};
```

Comment:
`StaticStrings::DataSourceType` seems to be an `std::string` already, so the extra `std::string{...}` wrapper here seems to be unnecessary.
But isn't this prone to the static initialization init order fiasco? What if the data in the StaticStrings.cpp TU gets initialized later than the one in this TU?
```suggestion
LogTopic TOPIC{std::string{StaticStrings::DataSourceType}, LogLevel::INFO};
```

Reply:
@jsteemann `StaticStrings::DataSourceType` is `std::string_view`, see `IResearchCommon.h:75`

Reply:
I guess initialization order should be fine here given it is static constexpr string_view

Reply:
Yes, my mistake. I wasn't aware of anything in StaticStrings that is an `std::string_view`...